### PR TITLE
Connect clientside to serverside to retrieve user info upon signup

### LINKED ISSUE #13: Connect settings page to backend to pull data from database
- setup /users route in serverside and create query to retrieve and update user information

### LINKED ISSUE #19: Create view for settings/profile information page
At a minimum, they must log the following data elements:
- Date and Time of Event
- Patient Identification
- User Identification
- Type of Action (additions, deletions, changes, queries, print, copy)
- Identification of the Patient Data that is Accessed

Go here for more info:
http://ehrscience.com/2013/01/28/ehr-design-basics-tracking-data-changes-and-accesses/

### LINKED ISSUE #34: Setup react/redux in frontend to connect to server

### PR DESCRIPTION
**Pull request by teganduong**
- closes #13 
- closes #19 
- closes #34 
- reformats signup to a page rather than a modal
- redirects to dashboard after user signs up

### CODE REVIEW

File: src/server/controllers/usersCtrl.js [hunk 1]
```diff
@@ -2,34 +2,23 @@ const db = require('../db/connection.js');
...
+  db.none('insert into users(name, username, password, email, weight, bmi, goal)' +
+      'values(${name}, ${username}, ${password}, ${email}, ${weight}, ${bmi}, ${goal})',
     req.body)
     .then(() => {
       res.status(201)
         .json({
           status: 'success',
-          message: 'Inserted user into db'
+          message: 'Inserted user into db',
+          data: req.body
```
Comment (ajgrande924) [inline src/server/controllers/usersCtrl.js:14]: need to send the user_id along with the user object, i think the command is 'returning id' after you insert the values

File: src/server/db/schema.sql [hunk 2]
```diff
@@ -1,18 +1,7 @@
...
 
--- DROP TABLE IF EXISTS users_teams;
--- DROP TABLE IF EXISTS users_daily_activities;
--- DROP TABLE IF EXISTS users_incentives;
--- DROP TABLE IF EXISTS challenges;
--- DROP TABLE IF EXISTS users;
--- DROP TABLE IF EXISTS daily_activities;
--- DROP TABLE IF EXISTS incentives;
--- DROP TABLE IF EXISTS teams;
-
```
Comment (ajgrande924) [inline src/server/db/schema.sql:18]: keep the drop table commands (line 7-14) so if we have to reseed if any of the table properties change we can

File: src/client/index.html [hunk 3]
```diff
@@ -4,19 +4,14 @@
...
     <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
-    
-	  <!-- jQuery -->
+
     <script src="assets/lib/jquery-2.2.4.min.js"></script>
-    
-    <!-- Bootstrap JS -->
     <script src="assets/lib/bootstrap.min.js"></script>
     
-	  <!-- Custom CSS -->
```
Comment (ajgrande924) [inline src/client/index.html:19]: why are we removing comments? dont these make sense to improve the readability of our code?

File: package.json [hunk 4]
```diff
@@ -47,15 +45,14 @@
     "express": "~4.13.1",
     "express-session": "^1.13.0",
     "isomorphic-fetch": "^2.2.1",
-    "mongoose": "^4.5.0",
-    "nodemon": "^1.9.2",
```
Comment (ajgrande924) [inline package.json]: we will need nodemon when we deploy using a docker container, this was used during the demo

Comment (ajgrande924) [inline package.json]: nevermind it was redundant, this is fun commenting

File: .gitignore [hunk 5]
```diff
@@ -10,6 +10,9 @@ pids
 *.pid
 *.seed
 
+# API Keys
+src/client/api/google-api-key.js
```
Comment (ajgrande924) [inline .gitignore]: can we rename this to config.js or api-keys.js so if/when we want to add new api keys we can just add them to that file. Also if this is modified, can the person modifying send the file with the api keys as a snippet on slack so we can get the update.

File: src/client/assets/css/Profile.css [hunk 6]
```diff
@@ -10,11 +10,9 @@
 
 .top-container {
 	width: 100%;
-	border-width: 0.5px;
```
Comment (ajgrande924) [inline src/client/assets/css/Profile.css:13]: keep these commented as opposed to deleting (line 13-15) every time they show up

Comment (teganduong) [inline src/client/assets/css/Profile.css:13]: - no, these lines of code make the styling ugly so it's unnecessary and should be removed

File: src/server/controllers/usersCtrl.js (hunk 1 above)
Comment (teganduong) [inline src/server/controllers/usersCtrl.js:14]: - i just sent the whole user object back but user_id shouldn't be necessary, all we need is the username to fetch info (as long as we make the username field unique), especially when we use external apis

File: .gitignore (hunk 5 above)
Comment (lynnc24) [inline .gitignore]: can you please consolidate my config file with yours so there's only one?  you can delete the path for the config file under gitinore.

File: src/client/components/Dashboard/SideNavBar.js [hunk 7]
```diff
@@ -1,21 +1,22 @@
...
     <ul className="sidebar-nav">
-      <li><Link to="/dashboard">My Dashboard</Link></li>
-      <li><Link to="/dashboard/teams">Teams</Link></li>
-      <li><Link to="/dashboard/challenges">Challenges</Link></li>
-      <li><Link to="/dashboard/goals">Goals</Link></li>
-      <li><Link to="/dashboard/stats">Stats</Link></li>
-      <li><Link to="/dashboard/log">Log</Link></li>
-      <li><Link to="/dashboard/tips">Tips</Link></li>
-      <li><Link to="/settings">Settings</Link></li>
-      <li><Link to="/home">Signout</Link></li>
```
Comment (pchim) [inline src/client/components/Dashboard/SideNavBar.js:16]: Would we still need a link to signout at the bottom of the nav bar?

File: src/client/components/Dashboard/SideNavBar.js [hunk 8]
```diff
@@ -1,21 +1,22 @@
-import React, { Component } from 'react';
+import React, { PropTypes } from 'react';
 import { Link } from 'react-router';
 
-const SideNavBar = () => (
+const SideNavBar = ({ username }) => (
   <div id="sidebar-wrapper">
     <p id="sidebar-title">Fit Together</p>
     <ul className="sidebar-nav">
-      <li><Link to="/dashboard">My Dashboard</Link></li>
```
Comment (pchim) [inline src/client/components/Dashboard/SideNavBar.js:8]: I think we'd need a My Dashboard link to the front page of the their profile

**Pull request merged**